presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -217,7 +218,7 @@ public DynamicFilter createDynamicFilter(QueryId queryId, List<DynamicFilters.De
                     return NOT_BLOCKED;
                 }
 
-                return toCompletableFuture(whenAnyComplete(undoneFutures));
+                return unmodifiableFuture(toCompletableFuture(whenAnyComplete(undoneFutures)));
```

Comment:
please add a test. Please also update `io.prestosql.sql.planner.LocalDynamicFiltersCollector.TableSpecificDynamicFilter#isBlocked` + test

Reply:
Done in https://github.com/prestosql/presto/pull/5099